### ISSUE TITLE
Failing on-chain verification of proofs containing ecdsa signature verification

### ISSUE THREAD

**Issue by signorecello** [noir-lang/noir#3245]
### Aim

My goal is to prove an ECDSA signature verifies, with a circuit such as:

```rust
fn main(
  pub_key_x: [u8; 32],
  pub_key_y: [u8; 32],
  signature: [u8; 64],
  hashed_message: pub [u8; 32],
) {
    let valid_signature = verify_signature(pub_key_x, pub_key_y, signature, hashed_message);
    assert(valid_signature);
}
```

### Expected Behavior

I expect the circuit to produce a proof that can be verified both on-chain and off-chain.

### Bug

Proof verifies off-chain, but fails to verify on-chain with `PROOF_FAILURE`

### To Reproduce

1. Clone this MRE repo: https://github.com/signorecello/noir-breaking-verify-signature
2. Use nargo version 0.17.0 (previous versions break too, at least since 0.5.x)
3. Run `yarn` and `yarn compile`
4. Run `yarn test`

### Installation Method

None

### Would you like to submit a PR for this Issue?

No

**Comment by Savio-Sou** [comment]
For additional context: https://github.com/AztecProtocol/aztec-packages/issues/1844

**Comment by signorecello** [comment]
Good call, totally forgot this was already reported lol. @kevaundray it seems to point to bb.js as the cause, not Noir...

**Comment by signorecello** [comment]
Testing this with the outputs of `nargo prove` and `nargo codegen-verifier` and using remix to run them, I can confirm they verify correctly.

However, running the tests in the repo above fails with `PROOF_FAILURE`, as mentioned

Just updated the above repo to reflect this

**Comment by Savio-Sou** [comment]
In that case maybe it's also worth testing:
- Prove in JS -> verify in Remix
- `nargo prove` -> verify "on-chain" in JS

**Comment by valeryz** [comment]
friendly bump, also ran into this same problem.  A friendly Aztec dev at  Istanbul devconnect pointed to this issue :)

**Comment by signorecello** [comment]
Thanks @valeryz this is high prio together with recursive proofs. But it's quite a lot of work as it touches the depths of barretenberg, so it could take a bit

**Comment by signorecello** [comment]
Solved in 0.19.4 🔥

**Comment by Turupawn** [comment]
I :heart: U Pedro thank you
Will test this now
